release_dashboard/test/fakes/fake_clean_context.dart

```diff
@@ -0,0 +1,31 @@
+// Copyright 2014 The Flutter Authors. All rights reserved.
+// Use of this source code is governed by a BSD-style license that can be
+// found in the LICENSE file.
+
+import 'package:conductor_core/conductor_core.dart';
+import 'package:file/file.dart';
+
+/// Initializes a fake [CleanContext] in a fake local environment for testing.
+///
+/// [runOverride] parameter overrides the parent's [run] method if it is not null. Else the parent's
+/// [run] method is used.
+class FakeCleanContext implements CleanContext {
+  FakeCleanContext({
+    this.runOverride,
+  });
+
+  /// An optional override async callback for the real [run] method.
+  Future<void> Function()? runOverride;
```

Comment:
Instead of an override, we can simplify by making this an `Exception` and if it's passed throw it.

Reply:
In the tests, `runOverride` is also receiving a callback here: https://github.com/Yugue/cocoon/blob/fabe9178a7025a9656b21baa95f16079349e9357/release_dashboard/test/widgets/clean_release_button_test.dart#L91

Would it be ok to keep as a `runOverride` callback? In that case, different callbacks could be passed such as adding a delay. It also makes the fake more flexible to accommodate for future changes.

Reply:
SG